storagenode/contact/endpoint.go

```diff
@@ -14,24 +15,33 @@ import (
 
 	"storj.io/storj/pkg/identity"
 	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
 )
 
 // Endpoint implements the contact service Endpoints
 type Endpoint struct {
-	log     *zap.Logger
-	service *Service
+	log       *zap.Logger
+	pingStats *PingStats
+}
+
+// PingStats contains information regarding who and when the node was last pinged
+type PingStats struct {
```

Comment:
why do we need to track when/who the node was last pinged?

Reply:
just for the node dashboard

Reply:
I believe the dashboard indicates the last time you were contacted to let you know you are still externally accessible